### PR TITLE
🌺 Quality of life improvements

### PR DESCRIPTION
**Pull request by erunion**
rdme as it exists has a horde of usability issues:

* No commands are documented.
* There's no way to get help for individual commands (and they aren't documented anyway).
* Error handling is all over the place. Sometimes we return nice, colored, errors, others we dump raw JSON responses to the console.

There's **got** to be a better way!

1CDX

### Improvements
* [x] All commands have been fully documented.
* [x] More command categories/groups have been created: "Administration", "Sync Swagger/OpenAPI files", "Documentation", and "Versions"
* [x] Heavily improved error handling to account for the wide variety of error cases that might happen (API errors, type errors, bad command argument errors, unexpected errors, etc.)
* [x] Cleaned up a lot of the errors that we were returning so they're more consistently represented.
* [x] 🆕 Added support for getting help via `rdme <command> --help` and `rdme help <command>`
* [x] 🆕 Related commands are now surfaced on command help screens. So for example if you do `rdme help versions`, you'll see a related commands list of every other versions-related command available.
* [x] 🆕 Owlbert has been added to the main `rdme help` screen because everyone needs more Owlbert.

### In action
```
$ ./rdme.js

          ░                                        ░
          ░░                                       ░░
        ░░░░░                                     ░░░░ ░
       ░░░░░░░                                    ░░░░░░
       ▐░░░░░░▄                                 ░░░░░░░░░
       ░░░░░░░░░▐░░░                        ░░░░░░░░░░░░░   rdme
        ░░░░░░░░░░░░░░░░░▐▐▐░░░░░░░░░░▐▐░░░░░░░░░░░░░░░░░░
      ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░    a utlity for interacting with ReadMe.io
      ▐░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▄▄▄▄░░░
      ▐░░░░░░░░░░░░░░░▄█▓▓▓▓▓▓█▄░░░░░░░░░░░░░░▄█▓▓▓▓▓▓▓▓▓▄░          ░░
      ▐░░░░░░░░░░░░▄▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄░░░░░░░░░▄▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄       ░░░░░░░░░
      ░░░░░░░░░░░░▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▌░░░░░░░▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄      ░░░░░░░░░░
      ░░░░░░░░░░░▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄░░░░░░▒▓▓▓▓█▀▓▓▓▓▓▓▓▓▓▓▓    ░░░░░░░░░░░░░░░░
      ░░░░░░░░░░░▒▓▓▓▓▀  ▓▓▓▓▓▓▓▓▓▓▓▌░░░░░░▐▓▓▓   ▀▀▓▓▓▓▓▓▓▓▌   ░░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░▐▓▓▓▌     ▐▓▓▓▓▓▓▓▓░░░░░░░░▀▓▓▄   ▄▓▓▓▓▓▓▓▓   ░░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░░▐▓▓▓▌▄ ▄▄▓▓▓▓▓▓▓▓▀░░░░░░░░░░▀▓▓▓▓▓▓▓▓▓▓█▀  ░░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░░░░▀▓▓▓▓▓▓▓▓▓▓▓█▀░░░░░░░░░░░░░░░▀▀▀▀▀▀░░░  ░░░░░░░░░░░░░░░
      ▐░░░░░░░░░░░░░░░░▀▀▀▀▀▀▀░░░░░░░░░░▄▄▄░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
      ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▒▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░
     ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▀▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░
    ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▀░░░░░░░░░░░░░░░░░░░░░
   ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░

Usage: rdme <command> [arguments]

Available commands

  Administration
    $ rdme login             Login to a ReadMe project. 

  Sync Swagger/OpenAPI files
    $ rdme swagger           Upload, or sync, your Swagger/OpenAPI file to ReadMe. 

  Documentation
    $ rdme docs              Sync a folder of markdown files to your ReadMe project.         
    $ rdme docs:edit         Edit a single file from your ReadMe project without saving      
                             locally.                                                        

  Versions
    $ rdme versions          List versions available in your project or get a version by     
                             SemVer (https://semver.org/).                                   
    $ rdme versions:create   Create a new version for your project.                          
    $ rdme versions:delete   Delete a version associated with your ReadMe project.           
    $ rdme versions:update   Update an existing version for your project.                    

  Other useful commands
    $ rdme open              Open your current ReadMe project in the browser.                
    $ rdme oas               OAS related tasks. See https://npm.im/oas for more information. 

Options

  -h, --help       Display this usage guide      
  -v, --version    Show the current rdme version 

Run rdme help <command> for help with a specific command.

```

```
$ ./rdme.js help versions

List versions available in your project or get a version by SemVer (https://semver.org/).

Usage

  rdme versions [options] 

Options

  --key string       Project API key          
  --version string   Project version          
  -h, --help         Display this usage guide 

Related commands

  $ rdme versions:create   Create a new version for your project.                
  $ rdme versions:delete   Delete a version associated with your ReadMe project. 
  $ rdme versions:update   Update an existing version for your project.          

```

### CODE REVIEW

**Comment by erunion** [comment]
@gratcliff You should take a look at this to make sure I didn't lose anything you added in the versions additions.

**Comment by gratcliff** [comment]
> @gratcliff You should take a look at this to make sure I didn't lose anything you added in the versions additions.

Absolutely! In the processes of looking as we speak. Some of that `fs` work is top notch!

File: lib/help.js [hunk 1]
```diff
@@ -1,67 +1,149 @@
...
+  // http://asciiart.club
+  return `          ░                                        ░
+          ░░                                       ░░
+        ░░░░░                                     ░░░░ ░
+       ░░░░░░░                                    ░░░░░░
+       ▐░░░░░░▄                                 ░░░░░░░░░
+       ░░░░░░░░░▐░░░                        ░░░░░░░░░░░░░   ${config.cli.blue.bold}
+        ░░░░░░░░░░░░░░░░░▐▐▐░░░░░░░░░░▐▐░░░░░░░░░░░░░░░░░░
+      ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░    ${
+        `a utlity for interacting with ReadMe.io`.bold
```
Comment (kanadgupta) [inline lib/help.js]: We could probably drop the .io in _ReadMe.io_? @mjcuva could maybe confirm or deny if this is the direction we're heading in with the new homepage domain and all.

**Comment by amyshi188** [comment]
I noticed when importing a spec file with the CLI tool that the command is different:
Screen Shot 2019-07-24 at 12 23 16 PM

It doesn't indicate that the Swagger file needs to come after "swagger" which might be confusing to new users. It would be clearer if it looked something like this: `rdme swagger {path-to-swagger.json} --key={api-key}`. 

I don't know what the reason was for changing it but maybe this is something to consider?

**Comment by erunion** [comment]
@amyshi188 How about this instead:

```
$ ./rdme.js help swagger

Upload, or sync, your Swagger/OpenAPI file to ReadMe.

Usage

  rdme swagger [path-to-spec.json] 

Options

  --key string       Project API key                                                               
  --id string        Unique identifier for your specification. Use this if you're resyncing an     
                     existing specification                                                        
  --version string   Project version                                                               
  -h, --help         Display this usage guide                                                      

```

I kind of want to keep `--key` off the usage line for it because it's technically not required because if you've used `rdme login` we save your key and re-use that.

**Comment by gratcliff** [comment]
@erunion Reviewed it a couple days ago, sorry for the late reply. Looks great overall + it doesn't seem like you've implemented any breaking changes. `

File: lib/help.js (hunk 1 above)
Comment (mjcuva) [reply lib/help.js]: Yes! We are in the bold new era of being a .com and not needing that pesky tld everywhere

Comment (erunion) [reply lib/help.js]: I'll change this to just be "a utlity for interacting with ReadMe", and also fix the typo in "utility" while I'm at it.

**Comment by mjcuva** [comment]
@amyshi188 We actually changed it so you don't need the path to the file anymore! If you don't provide one it will automatically try to find one, and if it can't then we ask the user to enter the path. So for new users it's probably better to no include the path at all!

**Comment by erunion** [comment]
I'm good to keep the usage for the swagger command as `rdme swagger [spec]` since it'll be nice to know when looking for help on that command to see that it does take a spec as an argument even though it's optional.

I talked offline with @gratcliff and I've accidentally removed a heap of argument support on the new versions commands, so I'm working on resurrecting those.

**Comment by erunion** [comment]
@gratcliff @mjcuva Pushed up a handful of fixes for the `versions` commands that I broke, as well as updating `rdme versions` to return its data within a table:

```
$ ./rdme.js versions
┌─────────┬──────────┬───────────────┬───────────┬─────────┬───────────┬──────────────────────────┐
│ Version │ Codename │ Is deprecated │ Is hidden │ Is beta │ Is stable │ Created on               │
├─────────┼──────────┼───────────────┼───────────┼─────────┼───────────┼──────────────────────────┤
│ 1.0     │          │ yes           │ yes       │ yes     │ no        │ 2019-06-17T22:39:56.462Z │
└─────────┴──────────┴───────────────┴───────────┴─────────┴───────────┴──────────────────────────┘
```

**Comment by gratcliff** [comment]
> @gratcliff @mjcuva Pushed up a handful of fixes for the `versions` commands that I broke, as well as updating `rdme versions` to return its data within a table:
> 
> ```
> $ ./rdme.js versions
> ┌─────────┬──────────┬───────────────┬───────────┬─────────┬───────────┬──────────────────────────┐
> │ Version │ Codename │ Is deprecated │ Is hidden │ Is beta │ Is stable │ Created on               │
> ├─────────┼──────────┼───────────────┼───────────┼─────────┼───────────┼──────────────────────────┤
> │ 1.0     │          │ yes           │ yes       │ yes     │ no        │ 2019-06-17T22:39:56.462Z │
> └─────────┴──────────┴───────────────┴───────────┴─────────┴───────────┴──────────────────────────┘
> ```

I agree with you that the table is cleaner, but it's pruning some key information in regards to pulling specific versions: 
Screen Shot 2019-07-29 at 2 10 16 PM

**Comment by erunion** [comment]
@gratcliff That's fair, but I was thinking that since this is a CLI wrapper around the API, that everything should be "pretty" by default. Maybe we can add a `raw` option to the `versions` command that'll dump the raw API output instead?

I'll go back and work in the missing data for the single version that I'm not currently surfacing in the prettified version though.

**Comment by gratcliff** [comment]
> @gratcliff That's fair, but I was thinking that since this is a CLI wrapper around the API, that everything should be "pretty" by default. Maybe we can add a `raw` option to the `versions` command that'll dump the raw API output instead?
> 
> I'll go back and work in the missing data for the single version that I'm not currently surfacing in the prettified version though.

Works for me! If we do can you update the `README.md` to reflect that?

**Comment by erunion** [comment]
@gratcliff Took another pass on `versions` and it's ready for your review again.

```$ ./rdme.js versions --version=1.0
Version: 1.0
Codename: None
Created on: 2019-06-17T22:39:56.462Z
Released on: 2019-06-17T22:39:56.462Z
┌───────────────┬───────────┬─────────┬───────────┐
│ Is deprecated │ Is hidden │ Is beta │ Is stable │
├───────────────┼───────────┼─────────┼───────────┤
│ yes           │ yes       │ yes     │ no        │
└───────────────┴───────────┴─────────┴───────────┘
```

Since we don't have an API endpoint to pull information from `categories`, I'm opting (for now) to not expose that in the "pretty" format, but I did add a `--raw` option to just dump the API output to the terminal instead.

File: lib/help.js [hunk 2]
```diff
@@ -1,67 +1,149 @@
...
+  // http://asciiart.club
+  return `          ░                                        ░
+          ░░                                       ░░
+        ░░░░░                                     ░░░░ ░
+       ░░░░░░░                                    ░░░░░░
+       ▐░░░░░░▄                                 ░░░░░░░░░
+       ░░░░░░░░░▐░░░                        ░░░░░░░░░░░░░   ${config.cli.blue.bold}
+        ░░░░░░░░░░░░░░░░░▐▐▐░░░░░░░░░░▐▐░░░░░░░░░░░░░░░░░░
+      ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░    ${
+        `a utlity for interacting with ReadMe`.bold
```
Comment (rafegoldberg) [inline lib/help.js]: This is a _ridiculously_ minor detail so feel free to ignore/punch me, but... we should adjust these lines so the text is aligned flush left. (Literally only off by a single space, I know, but my OCD just cannot abide it!)

Comment (rafegoldberg) [reply lib/help.js]: @erunion well that was extremely quick...

Comment (erunion) [reply lib/help.js]: gotta go fast 🏃💨

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # `rdme` - ReadMe's CLI
```
Comment (gratcliff) [inline README.md]: @erunion can you update the versions docs to reflect `--raw`? Looks great otherwise!! Happy to approve after that :)

**Pull request merged**